python/ray/air/checkpoint.py

```diff
@@ -48,6 +49,12 @@ class Checkpoint:
     There are no guarantees made about compatibility of intermediate
     representations.
 
+    For checkpoints which were created from dictionaries and then serialized into
+    directories, if additional files are added to the directory, these will be
+    serialized if the checkpoint is subsequently turned into a dictionary again.
+    In this case, the checkpoint dictionary will contain an additional private field,
+    ``_ray_additional_checkpoint_files``, containing the serialized files.
```

Comment:
Could you describe the behavior in a way that doesn't require users to poke into _ray_additional_checkpoint_files?
Ideally they shouldn't right?
How about let's say dict --> directory (add some more files) --> dict --> directory won't lose the added files?

Also maybe present this paragraph and the next in a coherent way. (they are pretty much the same thing)